tests/conftest.py

```diff
@@ -15,14 +34,122 @@
 
 @pytest.fixture(scope="session")
 def config():
-    yield get_config()
+    config = get_config()
+    config.is_test_mode = True
+    yield config
+
+
+@pytest.fixture
+def db(config):
+    """Yield a connection to the test DB."""
+    # Create the test DB enginge
+    assert config.is_test_mode
+    engine = get_db_engine(
+        database_uri=config.database.SQLALCHEMY_DATABASE_URI,
+    )
+
+    if not database_exists(engine.url):
+        logger.debug("Creating database at: %s", engine.url)
+        create_database(engine.url)
+
+    # Create the database tables
+    Base.metadata.create_all(engine)
+
+    SessionLocal = get_db_session(config, engine=engine)
+    session = SessionLocal()
+    yield session
+
+    session.close()
+    Base.metadata.drop_all(bind=engine)
+
+    engine.dispose()
 
 
 @pytest.fixture(autouse=True, scope="session")
 def env_vars():
     os.environ["TESTING"] = "True"
 
 
+@pytest.fixture(scope="session")
+def client():
+    app = FastAPI()
+    app.include_router(router)
+    with TestClient(app) as client:
+        yield client
+
+
 @pytest.fixture
 def fides_toml_path():
     yield ROOT_PATH / "fides.toml"
+
+
+@pytest.fixture
+def oauth_client(db):
+    """Return a client for authentication purposes."""
+    client = ClientDetail(
+        hashed_secret="thisisatest",
+        salt="thisisstillatest",
+        scopes=SCOPES,
+    )
+    db.add(client)
+    db.commit()
+    db.refresh(client)
+    yield client
+    client.delete(db)
+
+
+@pytest.fixture
+def auth_header(request, oauth_client, config):
+    client_id = oauth_client.id
+
+    payload = {
+        JWE_PAYLOAD_SCOPES: request.param,
+        JWE_PAYLOAD_CLIENT_ID: client_id,
+        JWE_ISSUED_AT: datetime.now().isoformat(),
+    }
+    jwe = generate_jwe(json.dumps(payload), config.security.APP_ENCRYPTION_KEY)
+
+    return {"Authorization": "Bearer " + jwe}
```

Comment:
Neat, I like how you're better making use of pytest capabilities with your `@pytest.mark.parametrize("auth_header", [[USER_CREATE]], indirect=True)` usage, instead of creating a function that returns a function like we do in fidesops.